src/app/shared/http.service.ts

```diff
@@ -46,12 +46,21 @@ export class HttpService extends Http {
 
   private catchRequestError () {
     return (res: Response) => {
-      if (res.status === 403 || ( res.status === 401 && !res.headers.has("www-authenticate"))) {
+      if (res.status === 403 || this.isAuthenticationError(res)) {
         this.broadcaster.broadcast('authenticationError', res);
       } else if (res.status === 500) {
         this.broadcaster.broadcast('communicationError', res);
       }
       return Observable.throw(res);
     };
   }
+
+  private isAuthenticationError(res: Response): boolean {
+    if (res.status === 401) {
+      const json: any = res.json();
+      return json && Array.isArray(json.errors) &&
+          json.errors.filter((e: any) => e.code === 'jwt_security_error').length >= 1;
```

Comment:
Note: This does not handle 2 corner cases 

1. The backend returns a 401 in cases where the jwt is valid but the associated identityID does not exist in the system. In general, this scenario shouldn't happen.
2. For a deprovisioned user, the jwt is valid, but the backend returns a 401 for API calls which do 'writes'. Example if a deprovisoined user tries to create a space, 401 is returned. So the changes in this PR wouldn't trigger a logout anymore for a deprovisioned user.

Having said the above, I think the changes in this PR is a good start.

Reply:
@sbose78 making a note on a PR will get lost. Do you want this done now or do you want to make an issue to track it?

Reply:
Thanks @joshuawilson . I've tracked it here https://github.com/openshiftio/openshift.io/issues/3736 
The PR can go in without these.